src/browser/browser.zig

```diff
@@ -380,7 +364,37 @@ pub const Page = struct {
 
         var response = try request.sendSync(.{});
         const header = response.header;
-        try self.session.cookie_jar.populateFromResponse(self.uri, &header);
+        try self.session.cookie_jar.populateFromResponse(request.uri, &header);
+
+        // update uri after eventual redirection
+        var buf: std.ArrayListUnmanaged(u8) = .{};
+        defer buf.deinit(arena);
```

Comment:
We can treat the allocator like an `ArenaAllocator` or like an `std.mem.Allocator`.

If we treat it like an `Arena`, we can remove unnecessary frees, which has a super-tiny performance cost.

If we treat it like an `Allocator`, we gain some flexibility with respect to changing the implementation.

It isn't a system-wide decisions, there are places where we'll have an arena and want to treat it like a generic allocator, and there are places where we'll definitely want to take advantage of the arena's behavior. The reason I like to call these ArenaAllocators `arena` is that it's clear to the writer & reader what the behavior is.

TL;DR - The `deinit` is almost certainly a no-op and could be removed. It's up to you.